lib/queryRelatedRecords.js

```diff
@@ -0,0 +1,52 @@
+const { getCollectionCrs, getGeometryTypeFromGeojson } = require('./helpers')
+const { computeFieldObject } = require('./field')
+
+module.exports = queryRelatedRecords
+
+function queryRelatedRecords (data, params = {}) {
+  const response = {
+    relatedRecordGroups: []
+  }
+
+  if (!params.returnCountOnly) response.fields = computeFieldObject(data, 'query', params)
+
+  const geomType = getGeometryTypeFromGeojson(data)
+  if (geomType) {
+    response.geomType = geomType
+    response.spatialReference = getCollectionCrs(data)
+    response.hasZ = false
+    response.hasM = false
+  }
+
+  if (data.features) {
+    response.relatedRecordGroups = data.features.map(recordGroupJSON => {
+      const recordGroup = {
+        objectId: recordGroupJSON.properties.objectid
+      }
+
+      if (params.returnCountOnly) {
+        // allow for preprocessing of count within provider
+        if ('count' in recordGroupJSON.properties) {
```

Comment:
The `in` syntax is pretty old and not consistent with the rest of the code base.  Once you do the destructuring I noted in other comment, you can just convert to `if (properties.count)`.